charts/opentelemetry-collector/templates/daemonset.yaml

```diff
@@ -6,6 +6,7 @@ metadata:
   namespace: {{ template "opentelemetry-collector.namespace" . }}
   labels:
     {{- include "opentelemetry-collector.labels" . | nindent 4 }}
+    {{- include "opentelemetry-collector.component" . | nindent 4 }}
```

Comment:
The `opentelemetry-collector.labels` already sets `app.kubernetes.io/component`. `opentelemetry-collector.component` sets the old-style `component` label. I don't think we need it

Reply:
if we don't include "opentelemetry-collector.component", the selector will not work:
```yaml
  selector:
    matchLabels:
      {{- include "opentelemetry-collector.selectorLabels" . | nindent 6 }}
      {{- include "opentelemetry-collector.component" . | nindent 6 }}
```